### PR TITLE
added console warning for non-generated keys.

### PR DESCRIPTION
**Pull request by Oripi**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,14 +7,63 @@ var through = require('through2'),
 
 var PLUGIN_NAME = 'gulp-json-angular-translate';
 
+function difference(object, base) {
+	function changes(object, base) {
```
Comment (dorduch) [inline index.js:11]: why the nested function?

File: index.js [hunk 2]
```diff
@@ -7,14 +7,63 @@ var through = require('through2'),
...
     return _.merge(prev, key.split('.').reduceRight(function (prev, curr) {
       var obj = {};
       obj[curr] = prev;
       return obj;
     }, json[key]));
   }, {});
+
+  const originalObjectLength = Object.keys(json).length;
+
+  const countKeys = obj => Object.keys(obj).reduce(((acc, curr) => {
```
Comment (dorduch) [inline index.js:53]: this function can be outside of `unflatten` scope

File: index.js [hunk 3]
```diff
@@ -7,14 +7,63 @@ var through = require('through2'),
...
+          ...acc,
+          [key]: obj[key],
+        };
+      }
+
+      const flattenedChild = flatten(obj[key], delimiter);
+
+      return {
+        ...acc,
+        ...Object.keys(flattenedChild).reduce((childAcc, childKey) => ({ ...childAcc, [`${key}${delimiter}${childKey}`]: flattenedChild[childKey] }), {}),
```
Comment (dorduch) [inline index.js:35]: it's not so clear to me what's happening here, maybe extract to smaller functions?

File: index.js (hunk 2 above)
Comment (ronnyrin) [reply index.js:53]: also don't mix arrow functions and native functions. we don't know on which node version this will run

File: index.js (hunk 1 above)
Comment (Oripi) [reply index.js:11]: this is what happens when you copy paste stuff :)

File: index.js (hunk 2 above)
Comment (Oripi) [reply index.js]: moved to outer function

File: test/index.js [hunk 4]
```diff
@@ -12,20 +12,48 @@ describe('gulp-json-angular-translate', function () {
...
+        stream.on('error', function (result) {
+          expect(result).to.deep.equal(
+            new Error(
+              'Found mismatch between generate and received keys. please see warnings above for more details'
+            )
+          );
+          done();
+        });
+        stream.on('data', function () {
+          expect(false).to.equal(true);
```
Comment (dorduch) [inline test/index.js:37]: I think you can use the `fail()` function

**Pull request merged**